pkg/controller/utils/machine_helper_test.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	err = codec.DecodeProviderSpec(&sampleMachine.Spec.ProviderSpec, awsconfig)
+	if err != nil {
+		t.Fatalf("Can't decode sample Machine ProviderSpec %v", err)
+	}
+
+	if len(awsconfig.LoadBalancers) != 1 {
+		t.Fatalf("Expected to have only 1 LoadBalancer, but got %d", len(awsconfig.LoadBalancers))
+	}
+	if awsconfig.LoadBalancers[0].Name != newLBList[0].Name {
+		t.Fatalf("Expected LB name to be %s, but got %s", newLBList[0].Name, awsconfig.LoadBalancers[0].Name)
+	}
+
+	if awsconfig.LoadBalancers[0].Type != newLBList[0].Type {
+		t.Fatalf("Expected LB type to be %s, but got %s", string(newLBList[0].Type), string(awsconfig.LoadBalancers[0].Type))
+	}
+}
+
+func TestAddAWSLBToMasters(t *testing.T) {
+	masterNames := make([]string, 3)
+	for i := 0; i < 3; i++ {
+		masterNames[i] = fmt.Sprintf("master-%d", i)
+	}
+	machineList, _ := testutils.CreateMachineObjectList(masterNames, "add-aws-elb", "master", testutils.DefaultRegionName, testutils.DefaultAzName)
+
+	objs := []runtime.Object{machineList}
+	mocks := testutils.NewTestMock(t, objs)
+
+	// each Machine ought to have 2 NLBs at the start, so let's check
+	for _, machine := range machineList.Items {
+		machineInfo := types.NamespacedName{
+			Name:      machine.GetName(),
+			Namespace: machine.GetNamespace(),
+		}
+		// reload the object to make sure we're not just working with the "in-memory"
+		// representation, that being, the un-saved one.
+		err := mocks.FakeKubeClient.Get(context.TODO(), machineInfo, &machine)
+		if err != nil {
+			t.Fatalf("Couldn't reload the test machine (named %s): %v", machineInfo.Name, err)
+		}
+		l, _, _, err := testutils.ValidateMachineLB(&machine)
+		if err != nil {
+			t.Fatalf("Couldn't lookup the LB info: %v", err)
+		}
+		if l != 2 {
+			t.Fatalf("Before the test we expect to have 2 load balancers, but got %d", l)
+		}
+	}
+
+	elbname := "myelb"
+	err := AddAWSLBToMasterMachines(mocks.FakeKubeClient, elbname, machineList)
+	if err != nil {
+		t.Fatalf("err %v", err)
+	}
+
+	// Now we should have 3 load balancers, so let's check -- and they should be NLBs
+	for _, machine := range machineList.Items {
+		machineInfo := types.NamespacedName{
+			Name:      machine.GetName(),
+			Namespace: machine.GetNamespace(),
+		}
+		// reload the object to make sure we're not just working with the "in-memory"
+		// representation, that being, the un-saved one.
+		err = mocks.FakeKubeClient.Get(context.TODO(), machineInfo, &machine)
+		if err != nil {
+			t.Fatalf("Couldn't reload the test machine (named %s): %v", machineInfo.Name, err)
+		}
+		l, _, lbTypes, err := testutils.ValidateMachineLB(&machine)
+		if err != nil {
+			t.Fatalf("Can't lookup LB info: %v", err)
+		}
+
+		if l != 3 {
+			t.Fatalf("After the test we expect to have 3 load balancers, but got %d", l)
+		}
+		for _, lbType := range lbTypes {
+			if lbType != awsproviderapi.NetworkLoadBalancerType {
+				t.Fatalf("Expected to have a NLB, but got %s instead", string(lbType))
+			}
```

Comment:
we may also want to look specifically for the LB named myelb here, with a found boolean